lib/common/errors.js

```diff
@@ -82,6 +82,12 @@ function ErrorFactory(util) {
     }
     util.inherits(TaskCancellationError, BaseError);
 
+    function TaskStopError(message) {
```

Comment:
A TaskCancellationError is an error caused by an external cancellation event that triggers downstream graph handling logic. A TaskStopError is a special error type only used internally to resolve conflicts when multiple task runners pick up the same task because of a heartbeat failure of one of the runners. In this case, we want to stop running the duplicate task but not persist any state or publish any events, since that will cause the graph to re-evaluate itself and potentially fail.